### PR TITLE
chore: Update CI actions

### PR DESCRIPTION
**Pull request by XhmikosR**
TODO:

- [ ] Adapt branch protection rules before the merge

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (styfle) [inline .github/workflows/ci.yml:1]: Let's keep the filename the same so we can see the diff here

**Comment by XhmikosR** [comment]
I kept the patches separate for this reason.

File: .github/workflows/ci.yml
Comment (XhmikosR) [reply .github/workflows/ci.yml:1]: I kept the patches selarate for this reason.

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -11,26 +11,25 @@ on:
...
       - name: Clone repo
         uses: actions/checkout@v4
-      - name: Use Node.js ${{ matrix.node }}
+        with:
+          persist-credentials: false
+      - name: Set up Node.js
         uses: actions/setup-node@v4
         with:
-          node-version: ${{ matrix.node }}
-      - run: corepack enable npm
```
Comment (styfle) [inline .github/workflows/nodejs.yml:32]: Please keep corepack so it runs the correct version of npm

Comment (XhmikosR) [reply .github/workflows/nodejs.yml:32]: Done. I also restored npm --version. I wasn't aware of this feature, you learn something new everyday :)

**Review by styfle — CHANGES_REQUESTED** [review]
Lets change this PR to adding dependabot only.

I don't like the other changes since its not clear the benefit.

**Comment by XhmikosR** [comment]
Well, there's no permissions specified (which could be even less later), newer actions versions, no more useless matrix and the use of `npm ci` and caching.

https://www.diffchecker.com/4IH8J4Kj/

Feel free to close if you only want the dependabot addition. I personally think the other changes are good.

**Pull request closed without merging**